components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipIterator.java

```diff
@@ -48,6 +48,11 @@
     public ZipIterator(Exchange exchange, InputStream inputStream) {
         this.exchange = exchange;
         this.allowEmptyDirectory = false;
+
+        if (inputStream == null) {
+            throw new IllegalStateException("Input stream cannot be null");
+        }
```

Comment:
I’m not sure that an IllegalStateException is the right choice in this case, I would rather use ´Objects.requireNonNull´ Instead.

Reply:
Indeed , changed to the proposed one.